contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@scroll-tech/contracts/L1/gateways/IL1GatewayRouter.sol";
+import "@scroll-tech/contracts/L1/rollup/IL2GasPriceOracle.sol";
+import "@scroll-tech/contracts/L1/IL1ScrollMessenger.sol";
+import "./interfaces/AdapterInterface.sol";
+
+/**
+ * @title Scroll_Adapter
+ * @notice Adapter contract deployed on L1 alongside the HubPool to facilitate token transfers
+ * and arbitrary message relaying from L1 to L2.
+ */
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+    uint32 public immutable l2GasLimit = 250_000;
```

Comment:
nit: why not use `constant` instead of `immutable`..?

Reply:
This is a great callout. I was following the stylization of these adapters:
- Base
- Boba
- Optimism

That said, I agree this is known at compile time so switching to constant would make more logical sense.

Done: https://github.com/across-protocol/contracts-v2/pull/381/commits/c1bb99414ce0854a8ca478035b5fe7e031647d21